src/app/services/http/transmart-http.service.ts

```diff
@@ -142,7 +143,7 @@ export class TransmartHttpService {
    */
   getCountsPerStudyAndConcept(constraint: Constraint): Observable<object> {
     const urlPart = 'observations/counts_per_study_and_concept';
-    const body = {constraint: TransmartConstraintMapper.mapConstraint(constraint)};
+    const body = {constraint: TransmartConstraintMapper.mapConstraintOnPatientLevel(constraint)};
```

Comment:
mapConstraint is supposed to be a generic method, why change it to a specific one?

Reply:
it is because most of the constraints have to be wrapped into patient subselection (except i.e. /query calls) .` mapConstraintOnPatientLevel` is basically `mapConstraint` wrapped into patient subselection.

Reply:
so how exactly does the sortIndex work?

Reply:
then we can still call it mapConstraint, right

Reply:
if the mark is 'subject', mapConstraint will wrap it with patient subselction, there is no need to hard code it

Reply:
when a mark is subject, it is wrapped into selected dimension subselection, which is not always patient. For some calls it has to stay like this, but for others there is a need for this additional layer of subselection (patient based).

Reply:
if we always need to provide the patient subselction wrapper, then maybe we shouldn't use 'mark' to denote it, since you always add that wrapper. Also, maybe we should rename 'mark' to 'dimension' as it makes more sense.

This is what I mean: if the dimension of constraint `c` is 'biomaterial', then 
`c.dimension = ConstraintDimension.BIOMATERIAL`

and when you try to convert `c` to a constraint object, you will wrap it with patient subselection, in a mapper e.g. :

`public toConstraintObject(c: Constraint): object {
    return {
        type: patient-subselection,
        constraint: c.toConstraintObject()
    }
}`

does this make sense?

Reply:
also, it seems that wrapping constraints with patient subselection on the top level is due to the technical requirements from the way transmart works, maybe sth. we can separate from how glowing bear converts constraint to objects?